Flickr/Sources/FlickrAppVCs/PhotoViewerVC.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  PhotoViewerVC.swift
+//  Flickr
+//
+//  Created by Ananth Desai on 28/03/22.
+//
+
+import Foundation
+import Nuke
+import UIKit
+
+class PhotoViewerVC: UIViewController {
+    private var url: URL
+    private var favoriteState: Bool
+    private var imageTitle: String?
+    private weak var stackView: UIStackView?
+    private weak var imageView: UIImageView?
+    private weak var favoriteButton: UIButton?
+
+    init(url: URL, imageTitle: String) {
+        self.url = url
+        self.imageTitle = imageTitle
+        if UserDefaults.standard.object(forKey: imageTitle) != nil {
+            favoriteState = true
+        } else {
+            favoriteState = false
+        }
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = viewBackgroundColor
+        let imageViewConstraints = setupImageView()
+        let stackViewConstraints = setupStackView()
+        let labelConstraints = returnLabel(stackView: stackView)
+        let iconConstraints = returnIcon(stackView: stackView)
```

Comment:
Why is there a need to pass stackview as a parameter here? Can't we access that directly from self in the function?